k-distribution/tests/regression-new/map-symbolic-tests-haskell/lookup

```diff
@@ -23,3 +23,29 @@ claim <k> lookup ( (X:MyId |-> 3 y |-> 4) [ x <- 5 ] [ X ] ) => . </k> requires
+claim <k> lookup ( (MAP:Map X:MyId |-> 1) [Y:MyId <- 2] [ Z:MyId ] orDefault 0 ) => . </k> requires Y ==K X
+claim <k> lookup ( (MAP:Map X:MyId |-> 1) [Y:MyId <- 2] [ Z:MyId ] orDefault 0 ) => . </k> requires Y ==K X andBool Z  ==K Y andBool Z  ==K X
+claim <k> lookup ( (MAP:Map X:MyId |-> 1) [Y:MyId <- 2] [ Z:MyId ] orDefault 0 ) => . </k> requires Y ==K X andBool Z =/=K Y andBool Z =/=K X
+claim <k> lookup ( (MAP:Map X:MyId |-> 1) [Y:MyId <- 2] [ Z:MyId ] orDefault 0 ) => . </k>
+claim <k> lookup ( (MAP:Map X:MyId |-> 1) [Y:MyId <- 2] [ Z:MyId ] orDefault 0 ) => . </k> requires Y =/=K X
+claim <k> lookup ( (MAP:Map X:MyId |-> 1) [Y:MyId <- 2] [ Z:MyId ] orDefault 0 ) => . </k> requires Y =/=K X andBool Z =/=K X
+claim <k> lookup ( (MAP:Map X:MyId |-> 1) [Y:MyId <- 2] [ Z:MyId ] orDefault 0 ) => . </k> requires Y =/=K X andBool Z =/=K Y
+claim <k> lookup ( (MAP:Map X:MyId |-> 1) [Y:MyId <- 2] [ Z:MyId ] orDefault 0 ) => . </k> requires Y =/=K X andBool Z  ==K Y andBool Z =/=K X
+claim <k> lookup ( (MAP:Map X:MyId |-> 1) [Y:MyId <- 2] [ Z:MyId ] orDefault 0 ) => . </k> requires Y =/=K X andBool Z =/=K Y andBool Z =/=K X
+claim <k> lookup ( (MAP:Map X:MyId |-> 1) [Y:MyId <- 2] [ Z:MyId ] orDefault 0 ) => . </k> requires Z  ==K X
+claim <k> lookup ( (MAP:Map X:MyId |-> 1) [Y:MyId <- 2] [ Z:MyId ] orDefault 0 ) => . </k> requires Z =/=K X
+claim <k> lookup ( (MAP:Map X:MyId |-> 1) [Y:MyId <- 2] [ Z:MyId ] orDefault 0 ) => . </k> requires Z  ==K Y
+claim <k> lookup ( (MAP:Map X:MyId |-> 1) [Y:MyId <- 2] [ Z:MyId ] orDefault 0 ) => . </k> requires Z =/=K Y
+claim <k> lookup ( (MAP:Map X:MyId |-> 1) [Y:MyId <- 2] [ Z:MyId ] orDefault 0 ) => . </k> requires Z =/=K Y andBool Z =/=K X
+claim <k> lookup ( (MAP:Map X:MyId |-> 1) [Y:MyId <- 2] [ Z:MyId ] orDefault 0 ) => . </k> requires Z =/=K Y andBool Z  ==K X andBool Y =/=K X
+claim <k> lookup ( (MAP:Map X:MyId |-> 1 Y:MyId |-> 2) [ Y:MyId ] orDefault 0 ) => . </k> requires X =/=K Y
+claim <k> lookup ( (MAP:Map Y:MyId |-> 1 X:MyId |-> 2) [ Y:MyId ] orDefault 0 ) => . </k> requires X =/=K Y
+claim <k> lookup ( (X:MyId |-> 1 MAP:Map Y:MyId |-> 2) [ Y:MyId ] orDefault 0 ) => . </k> requires X =/=K Y
+claim <k> lookup ( (Y:MyId |-> 1 MAP:Map X:MyId |-> 2) [ Y:MyId ] orDefault 0 ) => . </k> requires X =/=K Y
+claim <k> lookup ( (X:MyId |-> 1 Y:MyId |-> 2 MAP:Map) [ Y:MyId ] orDefault 0 ) => . </k> requires X =/=K Y
+claim <k> lookup ( (Y:MyId |-> 1 X:MyId |-> 2 MAP:Map) [ Y:MyId ] orDefault 0 ) => . </k> requires X =/=K Y
+claim <k> lookup ( (X:MyId |-> 3 y |-> 4) [ x <- 5 ] [ X ] orDefault 0 ) => . </k> requires X =/=K x
+claim <k> lookup ( (X:MyId |-> 3 y |-> 4) [ x <- 5 ] [ X ] orDefault 0 ) => . </k> requires X =/=K y
+claim <k> lookup ( ( (y |-> 4) [ x ] orDefault 0 ) ==K 4 ) => . </k>
+claim <k> lookup ( (Y:MyId |-> 1 X:MyId |-> 2) [ Y:MyId ] orDefault 0 ) => . </k>
+claim <k> lookup ( (X:MyId |-> 1 Y:MyId |-> 2) [ Z:MyId ] orDefault 0 ) => . </k> requires Z =/=K X andBool Z =/=K Y
```

Comment:
We should add a tests which have the `in_keys` predicate in the side-condition:

```
claim <k> lookup( M:Map [ X <- 1 ] [ Y ] orDefault 0 ) => . </k> requires COND1 andBool COND2
```

Where `COND1` ranges over `[X ==K Y, X =/=K Y]`, and `COND2` ranges over `[Y in_keys(M), notBool Y in_keys(M)]`.

As it stands, we only have one test that exercises the `orDefault` behavior, and only with a single-element map.

Reply:
I can add these tests, and will, but I wanted to note that, in fact, test 51 also exercises several of the default cases.